test/browsercontext-cookies.spec.ts

```diff
@@ -210,3 +210,46 @@ it('should not return cookies with empty value', async ({context, page, server})
   expect(cookies.length).toBe(0);
 });
 
+it('should return secure cookies based on HTTP(S) protocol', async ({context}) => {
+  await context.addCookies([{
+    url: 'https://foo.com',
+    name: 'doggo',
+    value: 'woofs',
+    secure: true
+  }, {
+    url: 'http://foo.com',
+    name: 'catto',
+    value: 'purrs',
+    secure: false
+  }]);
+  const cookies = await context.cookies('https://foo.com');
+  expect(cookies.sort((a, b) => a.name.localeCompare(b.name))).toEqual([{
```

Comment:
```suggestion
  expect(new Set([cookies])).toEqual(new Set([{
```
If I get you right, you want to have a sort independent toEqual, then you can use a Set for that.

Reply:
Ah yeah, good idea. Two of the other tests in the file also use a `.sort`, should I change those here too? I'd make this change  locally anyway, since your suggestion doesn't have a new closing paren for the second `Set` and `cookies` is already an iterable.

Reply:
I changed them all.